docs/design/persistent-volume-provisioning.md

```diff
@@ -0,0 +1,122 @@
+# Persistent Volume Provisioning
+
+This document proposes a model for dynamically provisioning [Persistent Volumes](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/persistent-storage.md)
+
+### tl;dr
+
+One new API kinds:
+
+A `PersistentVolumeController` (PVC) is a storage resource provisioned by an administrator.  PVCs do not have a namespace.  Like a ReplicationController maintains a minimum number of "replicas" of a pod in the system, so a PersistentVolumeController maintains a minimum number of "replicas" of a certain type/size of PersistentVolume available in the system.
+
```

Comment:
On Jul 14, 2015, at 2:45 PM, Mark Turansky <notifications@github.com> wrote:

In docs/design/persistent-volume-provisioning.md
<https://github.com/GoogleCloudPlatform/kubernetes/pull/6773#discussion_r34603920>
:

> @@ -0,0 +1,122 @@
> +# Persistent Volume Provisioning
> +
> +This document proposes a model for dynamically provisioning [Persistent Volumes](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/persistent-storage.md)
> +
> +### tl;dr
> +
> +One new API kinds:
> +
> +A `PersistentVolumeController` (PVC) is a storage resource provisioned by an administrator.  PVCs do not have a namespace.  Like a ReplicationController maintains a minimum number of "replicas" of a pod in the system, so a PersistentVolumeController maintains a minimum number of "replicas" of a certain type/size of PersistentVolume available in the system.
> +

@smarterclayton <https://github.com/smarterclayton> Having to make many
ReplicationControllers with Replicas=1 and 1 claim is a bad user experience.

if a user were to lay claim to a PersistentVolumeController -- which is a
creator and manager of its own pool of storage -- could we use this in
conjunction with a ReplicationController to scale something like MongoDB?
Different dev effort than this PR, but this PR's work could be steps
towards solving that scaling issue.



Possible.  The design of how we assign unique volumes to pods under a
replication controller (and reuse them when pods dies) is really important,
so we should try and get the discussion moving forward on that and reach
closure.  Nominal services are tied to this.


—
Reply to this email directly or view it on GitHub
<https://github.com/GoogleCloudPlatform/kubernetes/pull/6773/files#r34603920>
.